src/client/Microsoft.Identity.Client/ApiConfig/GetAuthorizationRequestUrlParameterBuilder.cs

```diff
@@ -84,6 +84,19 @@ public GetAuthorizationRequestUrlParameterBuilder WithExtraScopesToConsent(IEnum
             return this;
         }
 
+        /// <summary>
+        /// Used to secure authorization code grant via Proof of Key for Code Exchange (PKCE).
+        /// For more information, see the PKCE RCF:
+        /// https://tools.ietf.org/html/rfc7636
+        /// </summary>
+        /// <param name="codeVerifier">High-entropy cryptographic random STRING. If null, MSAL.NET will generate it.</param>
+        /// <returns></returns>
+        public GetAuthorizationRequestUrlParameterBuilder WithPkce(out string codeVerifier)
+        {
+            Parameters.CodeVerifier = codeVerifier = ServiceBundle.PlatformProxy.CryptographyManager.GenerateCodeVerifier();
+            return this;
```

Comment:
`codeVerifier ?? ServiceBundle.PlatformProxy.CryptographyManager.GenerateCodeVerifier();`

Reply:
should the user be able to specify their own code verifier here? if that is the case I can make this a ref instead of out

Reply:
oh yeah, i didn't even see the `out`...thanks  🤦